src/main/java/de/rub/nds/modifiablevariable/Logger/CustomLayout.java

```diff
@@ -0,0 +1,562 @@
+/*
+ * ModifiableVariable - A Variable Concept for Runtime Modifications
+ *
+ * Copyright 2014-2022 Ruhr University Bochum, Paderborn University, and Hackmanit GmbH
+ *
+ * Licensed under Apache License 2.0 http://www.apache.org/licenses/LICENSE-2.0
+ */
+package de.rub.nds.modifiablevariable.Logger;
+
+import de.rub.nds.modifiablevariable.util.ArrayConverter;
+import java.nio.charset.Charset;
+import java.util.*;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.DefaultConfiguration;
+import org.apache.logging.log4j.core.config.plugins.*;
+import org.apache.logging.log4j.core.impl.LocationAware;
+import org.apache.logging.log4j.core.layout.AbstractStringLayout;
+import org.apache.logging.log4j.core.layout.ByteBufferDestination;
+import org.apache.logging.log4j.core.layout.Encoder;
+import org.apache.logging.log4j.core.layout.PatternSelector;
+import org.apache.logging.log4j.core.pattern.LogEventPatternConverter;
+import org.apache.logging.log4j.core.pattern.PatternFormatter;
+import org.apache.logging.log4j.core.pattern.PatternParser;
+import org.apache.logging.log4j.core.pattern.RegexReplacement;
+import org.apache.logging.log4j.util.PropertiesUtil;
+import org.apache.logging.log4j.util.Strings;
+
+/**
+ * A layout for LOG messages in the correct format and also for logging ByteArrays with good
+ * performance. The CustomLayout is mostly copied from {@link
+ * org.apache.logging.log4j.core.layout.PatternLayout}.
+ */
+@Plugin(name = "CustomLayout", category = "Core", elementType = "layout", printObject = true)
+public class CustomLayout extends AbstractStringLayout {
```

Comment:
We should give this layout a more speaking name. `CustomLayout` seems to be a bit generic to me. What about `ModifiableVariableLayout` or similar?

Reply:
What do you think about `ExtendedPatternLayout`?

Reply:
Sounds good to me 👍